boltdesignsystem/bolt
# SVG Icon Theming / Coloring Fix

## Jira
http://vjira2:8080/browse/BDS-2012?filter=-1

## Summary
Globally addresses SVG color / theming issues (mostly seen in Firefox but a few issues in all browsers as well). Additionally addresses the "missing" background with the `file-small` via the flattened SVG file from @adamszalapski in #1746.

Before (left) vs After (right)
![image](https://user-images.githubusercontent.com/1617209/74773141-53d41480-525f-11ea-96c6-1ea869f5c2f7.png)

![image](https://user-images.githubusercontent.com/1617209/74773162-5fbfd680-525f-11ea-97d2-34073098898d.png)

![image](https://user-images.githubusercontent.com/1617209/74773185-6c442f00-525f-11ea-9cde-9669d8038b84.png)

![image](https://user-images.githubusercontent.com/1617209/74773216-7e25d200-525f-11ea-9583-228f80fdd0d8.png)

## Details
Most of the SVG icon color issues have to do with the CSS custom property syntax so this PR basically simplifies the syntax using only the original styling API values + removes some unnecessary cruft as a result.

While we need to confirm this, this PR might also eliminate the need for the updates in #1746 / http://vjira2:8080/browse/BDS-2023 since I suspect that these two are directly related.

## How to test
- Confirm tests passing as expected
- Review the SVG icon demos (especially in Firefox and Safari) to confirm the demos are now all rendering correctly
- It might also help to try out the demo code mentioned by @adamszalapski to confirm this style customization example is also working

```
<bolt-icon style="color: #7ACDCD; --bolt-theme-icon: #7ACDCD;" name="face-sad" size="xlarge"></bolt-icon>
```
